src/javaharness/java/arcs/api/PECInnerPortImpl.java

```diff
@@ -37,29 +42,42 @@ public void handleMessage(PortableJson message) {
                     // TODO: improve error handling.
                     throw new AssertionError("Cannot instantiate particle " + spec.name);
                 }
-                mapper.establishThingMapping(identifier, new Thing<NativeParticle>(particle));
+                mapper.establishThingMapping(messageBody.getString("identifier"), new Thing<NativeParticle>(particle));
                 break;
             case "DefineHandle":
-                StorageProxy storageProxy =
-                StorageProxyFactory.newProxy(
+                String identifier = messageBody.getString("identifier");
+                StorageProxy storageProxy = StorageProxyFactory.newProxy(
                     identifier,
                     TypeFactory.typeFromJson(messageBody.getObject("type")), messageBody.getString("name"), this);
                 mapper.establishThingMapping(identifier, new Thing<StorageProxy>(storageProxy));
                 break;
+            case "SimpleCallback":
+                // TODO: implement.
+                break;
             default:
                 throw new AssertionError("Unsupported message type: " + messageType);
         }
     }
 
     @Override
-    public void InitializeProxy(StorageProxy storageProxy) {
-        // TODO: Implement.
-        shellApi.postMessage("InitializeProxy: " + storageProxy.id);
+    public void InitializeProxy(StorageProxy storageProxy, Function<PortableJson, Void> callback) {
+        PortableJson message = jsonParser.parse("{}");
+        message.put("messageType", "InitializeProxy");
```

Comment:
just as an aside for future cleanup, in Java it is usually expected that string constants are pulled out into a public static final String CONSTANT = "constant" field and documented.

Reply:
declared constants for all the strings in this file. however for now they are private, so i'll document them later :)